### ISSUE TITLE
`hasActivityInThePast` false negative bug

### ISSUE THREAD

**Issue by lera222** [graphql/graphql-js#2242]
TL;DR

`hasActivityInThePast` returns `false`, despite the user having past activity.

Reproducible with [GraphQL Explorer](https://developer.github.com/v4/explorer/), using the query attached below:

1. first query with variables as follow:
```
{
  "username": "lera222",
}
```
This would query for a contribution collection starting a year ago, and ending today.
The result is:

```
{
  "data": {
    "rateLimit": {
      "remaining": 4999,
      "resetAt": "2019-10-29T16:50:59Z"
    },
    "user": {
      "contributionsCollection": {
        "hasAnyContributions": true,
        "hasActivityInThePast": false,
        "startedAt": "2018-10-28T04:00:00Z",
        "endedAt": "2019-10-30T03:59:59Z",

[....]
```
In particular, we have `"hasActivityInThePast": false,` and `"startedAt": "2018-10-28T04:00:00Z",`.

2. now use the same query, with the following variables:

```
{
  "username": "lera222",
  "end_at": "2018-10-28T04:00:00Z"
}
```

the result is:

```
{
  "data": {
    "rateLimit": {
      "remaining": 4999,
      "resetAt": "2019-10-29T16:55:06Z"
    },
    "user": {
      "contributionsCollection": {
        "hasAnyContributions": true,
        "hasActivityInThePast": false,
        "startedAt": "2017-10-28T04:00:00Z",
        "endedAt": "2018-10-28T04:00:00Z",
        "commitContributionsByRepository": [
          {
            "repository": {
              "id": "MDEwOlJlcG9zaXRvcnkxNTM3NzQ5Mjg="
            },
            "contributions": {
              "totalCount": 2
            }
          },
          {
            "repository": {
              "id": "MDEwOlJlcG9zaXRvcnkxNTM3NzQ3MDI="
            },
            "contributions": {
              "totalCount": 2
            }
          }
        ]
      }
    }
  }
}
```

specifically, we have `"hasAnyContributions": true,` (and the actual counts for two repos)

Query:

```
query ($username: String!, $end_at :DateTime, $start_at :DateTime) {
  
  rateLimit {
    remaining
    resetAt
  }

  user(login: $username) {

    contributionsCollection(from: $start_at, to: $end_at){

        hasAnyContributions
        hasActivityInThePast
        startedAt
        endedAt

        commitContributionsByRepository(maxRepositories:100){

            repository{
                id
            }

            contributions{
                totalCount
            }
        }
    }
  }
}


```

**Comment by lera222** [comment]
hey @IvanGoncharov. could you explain why you assigned a "question" label to this issue? it's a reproducible bug.

**Comment by Pepper-Wood** [comment]
For others who may be encountering this bug, the OP created the following Github help topic to continue investigation there: https://github.community/t/hasactivityinthepast-false-negative-bug/14431
